src/dmd/root/region.d

```diff
@@ -86,9 +89,28 @@ struct Region
      */
     bool contains(void* p)
     {
-        foreach (h; array[0 .. used])
+        if (!used)
+            return false;
+        if (!isSorted)
+            sort();
+
+        size_t low = 0;
+        if (p < array[low])
+            return false;
+
+        size_t high = array.length - 1;
+        if (array[high] + MaxAllocSize < p)
+            return false;
+
+        // Binary search
+        while (low <= high)
         {
-            if (h <= p && p < h + ChunkSize)
+            const mid = (low + high) >> 1;
```

Comment:
This looks like it might be vulnerable to overflow, but it isn't. `high` will always be less than half of `size_t.max` due to `array` having pointer-sized elements.

Reply:
right